internal/search/query/predicate.go

```diff
@@ -262,6 +264,56 @@ func (f *RepoContainsCommitAfterPredicate) Plan(parent Basic) (Plan, error) {
 	return ToPlan(Dnf(nodes))
 }
 
+// RepoDependenciesPredicate represents the `repo:dependencies(of: regex@rev)` predicate,
+// which filters to repos that are dependencies of the repos matching the given of regex.
+type RepoDependenciesPredicate struct {
+	Repo string
+}
+
+func (f *RepoDependenciesPredicate) ParseParams(params string) error {
+	nodes, err := Parse(params, SearchTypeRegex)
+	if err != nil {
+		return err
+	}
+
+	for _, node := range nodes {
+		if err := f.parseNode(node); err != nil {
+			return err
+		}
+	}
+
+	if f.Repo == "" {
+		return errors.New("repo must be set")
+	}
+
+	return nil
+}
+
+func (f *RepoDependenciesPredicate) parseNode(n Node) error {
+	switch v := n.(type) {
+	case Pattern:
+		f.Repo = v.Value
+		return nil
+	}
+
+	return errors.Errorf("unsupported node type %T", n)
+}
+
+func (f *RepoDependenciesPredicate) Field() string { return FieldRepo }
+func (f *RepoDependenciesPredicate) Name() string  { return "dependencies" }
+func (f *RepoDependenciesPredicate) Plan(parent Basic) (Plan, error) {
+	nodes := make([]Node, 0, 3)
+	nodes = append(nodes, Parameter{
+		Field: FieldDependencies,
+		Value: f.Repo,
+	})
+
+	nodes = append(nodes, nonPredicateRepos(parent)...)
+	return ToPlan(Dnf(nodes))
+}
```

Comment:
```suggestion
func (f *RepoDependenciesPredicate) Plan(Basic) (Plan, error) {
	return nil, nil
```

Do this now, since `Plan` doesn't matter for `repo:dependencies`